src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -619,6 +619,34 @@ RetWithError<UniquePtr<HashItf>> MbedTLSCryptoProvider::CreateHash(Hash algorith
     return {UniquePtr<HashItf>(Move(hasher)), ErrorEnum::eNone};
 }
 
+Error MbedTLSCryptoProvider::GenerateRandomString(String& result)
+{
+    size_t bytesNeeded = result.Size() / 2;
```

Comment:
Isn't it a typo? I expect Size to 0 in most cases
```suggestion
    size_t bytesNeeded = result.MaxSize() / 2;
```

Reply:
Please see test

Reply:
Updated